akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -62,7 +44,7 @@ final class Source[+Out, +Mat](private[stream] override val module: Module)
       new Source(
         module
           .growConnect(flowCopy, shape.outlet, flowCopy.shape.inlets.head, combine)
-          .replaceShape(SourceShape(flowCopy.shape.outlets.head)))
+          .replaceShape(SourceShape(flowCopy.shape.outlets.head))) // FIXME why is not .wrap() needed here?
```

Comment:
because you just grow the module. Wrapping would introduce a new hierarchy level. 

A little bit of history here:
First, I had a different type for "ModuleInProgress" which signified that it was not a final module, but a thing being built. While this was self-documenting, it turned out to be a huge duplication most of the time, so we dropped it. We just treat the current module as the "builder" implicitly now. Since Source is both a module, or a builder, this means that its submodule is being grown, or "finalized" if imported in a graph.

An example:
 - empty module: []
 - grow with map: [map]
 - grow with viaMat(someFlow):  [map -> someFlow]
 - wrap(): [[map -> someFlow]]
 - grow with filter: [[map -> someFlow] -> filter]

Another example:
 - one source: [single -> map ->]
 - one sink: [ -> filter -> foreach]
 - wiring them together: [single -> map -> filter -> foreach]

I.e. it tries to work on the same hierarchy level, growing the top-level module. If you use .named, it will wrap:
 - one source [single -> map ->]
 - adding a name: [named[single -> map ->] ->]
 - growing: [named[single -> map ->] -> filter ->]
 - one sink: [-> mapAsync -> foreach]
 - adding sink: [named[single -> map ->] -> filter -> mapAsync -> foreach]